README.md

```diff
@@ -1,125 +1,170 @@
-go generate ./...
-```
+## Configuration
 
-Execute test suite with statement coverage and save coverage data to `cov.txt`
-```shell
-go test ./... -coverprofile=cov.txt
-```
+Configuring Arcaflow Container Toolkit can be done in the `act.yaml` file as well as setting environment variables.
 
-## Build the carpenter
+### Configurable Variables
 
-* golang v1.18
-* current working directory is this project's root directory
-* flake8
-
-```shell
-go build carpenter.go
-```
+#### Required:
+  `IMAGE_NAME` Name of the image that Arcaflow Container Toolkit will build - string  
+  `IMAGE_TAG`  Tag of the image that Arcaflow Container Toolkit will build - string
+#### Required for Publishing to Quay:
+  `QUAY_USERNAME` Quay Username to be used for credentials - Default: ""  
+  `QUAY_PASSWORD` Quay Password to be used for credentials - Default: ""  
+  `QUAY_NAMESPACE` Quay Namespace to push image - Default: "" 
+#### Optional:    
+  `QUAY_CUSTOM_NAMESPACE` Quay Namespace to push image that is not QUAY_NAMESPACE - Default: ""  
+  `QUAY_IMG_EXP` Image label to automatically expire in Quay - Default: "never"  
+  `BUILD_TIMEOUT` Length of time before a build will fail in seconds - Default: 600  
 
-If successful, this will result in the arcaflow-plugin-image-builder executable, and it will be named `carpenter` in your current working directory.
+#### Additional Information
 
-## Build carpenter's image
+* `QUAY_IMG_EXP` more documentation and time formats can be found [here](https://docs.projectquay.io/use_quay.html#:~:text=Setting%20tag%20expiration%20from%20a%20Dockerfile)
+* `QUAY_CUSTOM_NAMESPACE` if set, will use in place of `QUAY_NAMESPACE`. More info [Arcaflow Container Toolkit and Reusable Workflows](#arcaflow-container-toolkit-and-reusable-workflows)
 
-### Requirements
+## Build Arcaflow Container Toolkit As Executable Locally
 
-* docker
-* current working directory is `arcaflow-plugin-image-builder`
+Arcaflow Container Toolkit can be ran locally by building an executable.  
+Configure `act.yaml` and or set environment variables.  
 
 ```shell
-docker build . --tag carpenter-img
+vi act.yaml
 ```
 
-## Example Build Execution
-
-### Requirements
-
-* arcaflow-plugin-image-builder executable named `carpenter`
-* `.carpenter.yaml` in the same directory as `carpenter` executable
-* flake8
-
-example `.carpenter.yaml`
+example `.act.yaml`
 ```yaml
 revision: 20220824
-image_name: arcaflow-plugin-template-python
-image_tag: '0.0.1'
-project_filepath: ../arcaflow-plugin-template-python
+image_name: "<IMAGE_NAME>"
+image_tag: "<IMAGE_TAG>"
+project_filepath: "<path/to/plugin/project/>"
 registries:
-  - url: ghcr.io
-    username_envvar: "GITHUB_USERNAME"
-    password_envvar: "GITHUB_PASSWORD"
   - url: quay.io
-    username_envvar: "QUAY_USERNAME"
-    password_envvar: "QUAY_PASSWORD"
-    namespace_envvar: "QUAY_NAMESPACE"
+    username_envvar: "<QUAY_USERNAME>"
+    password_envvar: "<QUAY_PASSWORD>"
+    namespace_envvar: "<QUAY_NAMESPACE>"
```

Comment:
We should probably note that these Quay values are secrets, not variables.

Reply:
This is the local config `act.yaml` file that is being used locally. This file would not be used by secrets in github. They would be free to fill this yaml file out and run ACT as an executable.